controllers/api/user.js

```diff
@@ -89,20 +89,173 @@ router.delete("/:userId", async (req, res) => {
         }
-        if(!req.body.password){
+        if (!req.body.password) {
             return res.sendStatus(400)
         }
-        if(await bcrypt.compare(req.body.password,user.password)){
+        if (await bcrypt.compare(req.body.password, user.password)) {
             const deleteUser = await User.destroy(
                 {
                     where: {
                         id: req.params.userId
                     }
                 })
             res.json(deleteUser)
-        }else{
-        res.sendStatus(403)
+        } else {
+            res.sendStatus(403)
+        }
+    } catch (err) {
+        console.log(err);
+        res.sendStatus(500);
+    }
+})
+
+//18
+router.get("/:userId/feed", async (req, res) => {
+    try {
+        const userId = 1;
+        const pageLength = parseInt(req.query["page-length"]) || 10;
+        const pageNumber = parseInt(req.query["page-number"]) || 0;
+
+        const user = await User.findByPk(req.params.userId, {
+            limit: pageLength,
+            offset: pageNumber * pageLength,
+            subQuery: false,
+            include: [{
+                model: Gallery,
+                as: "galleryFollowingUser",
+                include: [{
+                    model: Picture,
+                    attributes: ["id", "name", "S3URL"],
+                    include: [
+						{
+							model: User,
+							attributes: ["displayName"]
+						}, {
+							model: Comment,
+							attributes: ["id"]
+						}, {
+							model: Like,
+							attributes: ["id", "delta"]
+						}
+					]
+                }]
+            },{
+                through: {
+                    attributes: []
+                },
+                model: User,
+                as: "userFollowingUser",
+                attributes: ["id","displayName"],
+                include:[{
+                    model: Picture,
+                    attributes: ["id", "name", "S3URL"],
+                    include: [
+                        {
+                            model: Comment,
+                            attributes: ["id"]
+                        }, {
+                            model: Like,
+                            attributes: ["id", "delta"]
+                        }
+                    ]
+                }]
+            }]
+        })
+
+        const galleryPictures = user.galleryFollowingUser.map(picture => ({
+			id: picture.id,
+			name: picture.name,
+			commentCount: picture.comments.length,
+			imageURL: picture.S3URL,
+			score: picture.likes.reduce((score, { delta }) => score + delta, 0),
+			like: picture.likes.find(like => like.id === userId),
+		}));
+
+        const pictures = user.userFollowingUser.map(picture => ({
+			id: picture.id,
+			name: picture.name,
+			commentCount: picture.comments.length,
+			imageURL: picture.S3URL,
+			score: picture.likes.reduce((score, { delta }) => score + delta, 0),
+			like: picture.likes.find(like => like.id === userId),
+		}));
```

Comment:
`galleryFollowingUser` is an array of galleries, and `userFollowingUser` is an array of users. You need to `reduce()` them.

ie:
```
// Get every picture from every gallery:
const galleryPictures = user.galleryFollowingUser.reduce((gallery, pictures) => {
    // Concat the previous array of pictures to the next:
    return pictures.concat(user.galleryFollowingUser.map(picture => ({
        // Code from lines 166 -> 171 here
    }))
}, []); //<-- Start reducing with an empty array
```